libethashseal/EthashClient.cpp

```diff
@@ -57,6 +57,7 @@ EthashClient::EthashClient(ChainParams const& _params, int _networkID, p2p::Host
 
 EthashClient::~EthashClient()
 {
+    m_signalled.notify_all(); // to wake up the thread from Client::doWork()
```

Comment:
Why it must be in all derived classes?

Reply:
Because the derived classes' destructors override `~Client()`.

(Similarly, all drived class of `Worker` need to call `terminate()` in the destructors.)

Reply:
In case of destructors "override" does not mean that the destructor in the base class is replaced. The base class one will be called as the last one.

Reply:
`terminate()` must be called first, not the last.  Otherwise, the worker thread tries to continue working, using the most derived object.

Reply:
Agree. Is this the case also for `.notify_all()`?

Reply:
Yes, without the `notify_all()`, `terminate()` would block, waiting for the thread sleeping for one second.  For anything except `ClientTest`, I just reasoned this, not tried the behavior.